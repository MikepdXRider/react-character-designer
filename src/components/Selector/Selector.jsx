import React from 'react'
import Select from '../Select/Select.jsx'

export default function Selector({setHead, head}) {
    return (
        <section>
            <label>
                Head
                <Select
                    optionsArr={['cowboy', 'green', 'top', 'wizard']}
                    setState={setHead}
                    state={head}
                />
            </label>
        </section>
    )
}
