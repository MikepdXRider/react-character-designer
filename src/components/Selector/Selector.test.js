import {screen, render} from '@testing-library/react'
import Selector from './Selector.jsx'

it('renders selector component', () => {
    // render enables us to pseudo-render a component. This also lets us use the screen method later on
    const {container} = render(<Selector optionsArr={['test1', 'test2', 'test3']} />);
    // getByLabelText allows us to get the text within a <label> jsx tag.
    const headLabel = screen.getByLabelText('Head');
    const torsoLabel = screen.getByLabelText('Torso');
    const bottomLabel = screen.getByLabelText('Bottom');
    // expect the above elements to be in the rendered document.
    expect(headLabel).toBeInTheDocument();
    expect(torsoLabel).toBeInTheDocument();
    expect(bottomLabel).toBeInTheDocument();
    // expect container to match the produced snapshot(first test generates, all other test compare)
    expect(container).toMatchSnapshot();
})