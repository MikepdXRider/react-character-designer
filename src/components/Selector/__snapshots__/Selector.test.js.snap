// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders selector component 1`] = `
<div>
  <section>
    <label>
      Head
      <select>
        <option
          disabled=""
          hidden=""
          value=""
        >
          Please Choose...
        </option>
        <option
          data-testid="cowboy"
          value="cowboy"
        >
          cowboy
        </option>
        <option
          data-testid="green"
          value="green"
        >
          green
        </option>
        <option
          data-testid="top"
          value="top"
        >
          top
        </option>
        <option
          data-testid="wizard"
          value="wizard"
        >
          wizard
        </option>
      </select>
    </label>
    <label>
      Torso
      <select>
        <option
          disabled=""
          hidden=""
          value=""
        >
          Please Choose...
        </option>
        <option
          data-testid="cowboy"
          value="cowboy"
        >
          cowboy
        </option>
        <option
          data-testid="formal"
          value="formal"
        >
          formal
        </option>
        <option
          data-testid="jacket"
          value="jacket"
        >
          jacket
        </option>
      </select>
    </label>
    <label>
      Bottom
      <select>
        <option
          disabled=""
          hidden=""
          value=""
        >
          Please Choose...
        </option>
        <option
          data-testid="cowboy"
          value="cowboy"
        >
          cowboy
        </option>
        <option
          data-testid="jean"
          value="jean"
        >
          jean
        </option>
        <option
          data-testid="diaper"
          value="diaper"
        >
          diaper
        </option>
        <option
          data-testid="skirt"
          value="skirt"
        >
          skirt
        </option>
        <option
          data-testid="suit"
          value="suit"
        >
          suit
        </option>
      </select>
    </label>
    <label>
      Catchphrase
      <input
        type="text"
        value=""
      />
    </label>
    <button>
      Add Catchphrase
    </button>
  </section>
</div>
`;
